discord/interaction.py

```diff
@@ -0,0 +1,661 @@
+        if pong:
+            self.type = InteractionResponseType.pong
+        else:
+            if content is None and embeds is None:
+                self.type = InteractionResponseType.deferred_channel_message_with_source
+            else:
+                self.type = InteractionResponseType.channel_message_with_source
+            self.content = content
+            self.tts = tts
+            self.embeds = embeds
+            self.allowed_mentions = allowed_mentions
+            self.ephemeral = ephemeral
+
+    def to_dict(self):
+        result = {'type' : self.type.value}
+
+        data = {}
+        if self.content is not None:
+            data['content'] = str(self.content)
+        if self.tts != None:
+            data['tts'] = self.tts
+        if self.embeds is not None:
+            data['embeds'] = [embed.to_dict() for embed in self.embeds]
+        if self.allowed_mentions is not None:
+            data['allowed_mentions'] = self.allowed_mentions.to_dict()
+        if self.ephemeral:
+            data['flags'] = 64
+        if len(data) > 0:
+            result['data'] = data
+            
+        return result
+
+class InteractionMessage(Message):
+    """Represents a message sent through an interaction.
+
+    This allows you to edit or delete a message sent through an
+    interaction you received.
+
+    This inherits from :class:`discord.Message` with changes to
+    :meth:`edit` and :meth:`delete` to work.
+
+    """
+
+    __slots__ = ('_interaction')
+    
+    def __init__(self, *, interaction, data):
+        self._interaction = interaction
+        Message.__init__(self, state=interaction._state, channel=interaction.channel, data=data)
+
+    def edit(self, **fields):
+        """|coro|
+
+        Edits the message.
+
+        The content must be able to be transformed into a string via ``str(content)``.
+
+        Parameters
+        ------------
+        content: Optional[:class:`str`]
+            The content to edit the message with or ``None`` to clear it.
+        embeds: List[:class:`Embed`]
+            A list of embeds to edit the message with.
+        embed: Optional[:class:`Embed`]
+            The embed to edit the message with. ``None`` suppresses the embeds.
+            This should not be mixed with the ``embeds`` parameter.
+        allowed_mentions: :class:`AllowedMentions`
+            Controls the mentions being processed in this message.
+            See :meth:`.abc.Messageable.send` for more information.
+
+        Raises
+        -------
+        HTTPException
+            Editing the message failed.
+        InvalidArgument
+            You specified both ``embed`` and ``embeds`` or the length of
+            ``embeds`` was invalid.
+        """
+        return self._interaction.edit_message(self.id, **fields)
+
+    def _delete_delay_sync(self, delay):
+        time.sleep(delay)
+        return self._state._webhook.delete_message(self.id)
```

Comment:
Why is there a sync version here?

Reply:
Because this code is a modified version of that found in webhook.py. The delay code should have been removed altogether though as it isn't called by anything.